packages/dds/merge-tree/src/client.ts

```diff
@@ -1126,11 +1052,17 @@ export class Client extends TypedEventEmitter<IClientEvents> {
         }
     }
 
-    getContainingSegment<T extends ISegment>(pos: number, op?: ISequencedDocumentMessage, localSeq?: number) {
-        const args = this.getClientSequenceArgsForMessage(op);
+    getContainingSegment<T extends ISegment>(pos: number, sequenceArgs?: { referenceSequenceNumber: number; clientId: number; }, localSeq?: number) {
+        const args = sequenceArgs ?? this.getClientSequenceArgsForMessage(undefined);
```

Comment:
You can probably destructure `args` into `{clientId, referenceSequenceNumber}` and then reference them like that in the next line.

Reply:
rather than overloading, another approach that could work would be to make this function require `sequenceNumber` even though it doesn't really need it. I feel like that's more confusing than just tightening the above function signature though.

Reply:
I guess I could be ok with inlining the extraction in this function as you've done; if this is the only place we don't need sequenceNumber it's probably fine. However, we should definitely be using long clientIds in all of client's APIs. So you'll need to do a translation here.